web/packages/teleterm/src/services/tshd/cloneableClient.ts

```diff
@@ -404,9 +404,17 @@ export class MockedUnaryCall<Response extends object>
     onrejected?: (reason: any) => TResult2 | PromiseLike<TResult2>
   ): Promise<TResult1 | TResult2> {
     if (this.error) {
-      // Despite this being an error branch, it needs to use Promise.resolve. Otherwise we'd get
-      // uncaught errors. See https://www.promisejs.org/implementing/#then
-      return Promise.resolve(onrejected(this.error));
+      if (typeof onrejected === 'function') {
+        try {
+          // Despite this being an error branch, it needs to use Promise.resolve. Otherwise we'd get
+          // uncaught errors. See https://www.promisejs.org/implementing/#then
+          return Promise.resolve(onrejected(this.error));
```

Comment:
Oh actually, it makes total sense that this needs to be `Promise.resolve(onrejected(…))`.

If you have something like this:

```javascript
Promise.reject().then(() => { /* onfulfilled */ }, () => { /* onrejected */)
```

and onrejected doesn't reject itself, the final promise you get is fulfilled, not rejected.